webapp/components/admin_console/storage_settings.jsx

```diff
@@ -202,6 +204,25 @@ export default class StorageSettings extends AdminSettings {
                     disabled={this.state.driverName !== DRIVER_S3}
                 />
                 <BooleanSetting
+                    id='amazonS3SignV2'
+                    label={
+                        <FormattedMessage
+                            id='admin.image.amazonS3SignV2Title'
```

Comment:
The word "enable" makes it sound like the system would start using either V2 or V4 as the situation required, but the code says it's always one or always the other, not both.

In the UI I think the situation would be more clear if admins had to choose either v2 or v4 from a list. In that case the title would be: **AWS S3 Signature Version:** and then a drop-down. The text for the drop-down would be *Signature Version 2 Signing Process* and *Signature Version 4 Signing Process*.

Similarly for config.json. Instead of having true/false, the setting should be strings, either "V2" or "V4", with "V4" the default.

For the help text, do admins who use AWS already know why they would change from V4 to V2, or do they need help deciding? If the former, then no help text is required.